src/core/compiler.jl

```diff
@@ -288,49 +229,144 @@ function build_model_info(input_expr)
 end
 
 """
-    translate_tilde!(model_info)
+    replace_vi!(model_info)
+
+Replaces @varinfo() expressions to the VarInfo instance.
+"""
+function replace_vi!(model_info)
+    ex = model_info[:main_body]
+    vi = model_info[:main_body_names][:vi]
+    ex = MacroTools.postwalk(x -> @capture(x, @varinfo()) ? vi : x, ex)
+    model_info[:main_body] = ex
+    return model_info
+end
+
+"""
+    replace_logpdf!(model_info)
+
+Replaces @logpdf() expressions to the VarInfo instance.
+"""
+function replace_logpdf!(model_info)
+    ex = model_info[:main_body]
+    vi = model_info[:main_body_names][:vi]
+    ex = MacroTools.postwalk(x -> @capture(x, @logpdf()) ? :($vi.logp) : x, ex)
+    model_info[:main_body] = ex
+    return model_info
+end
+
+"""
+    replace_sampler!(model_info)
 
-Translates ~ expressions to observation or assumption expressions, updating `model_info`.
+Replaces @sampler() expressions to the VarInfo instance.
 """
-function translate_tilde!(model_info)
+function replace_sampler!(model_info)
     ex = model_info[:main_body]
+    spl = model_info[:main_body_names][:sampler]
+    ex = MacroTools.postwalk(x -> @capture(x, @sampler()) ? spl : x, ex)
+    model_info[:main_body] = ex
+    return model_info
+end
+
+"""
+\"""
```

Comment:
What does this block do?

Reply:
This one supports `.~` syntax only if in Julia 1.1 and higher. Writing code with `.~` in Julia 1.0 gives a parsing error even if not run at all.

Reply:
I will add a comment.